### PR TITLE
Feature: Implement mount command inside the namespace

### PR DESCRIPTION
**Pull request by Strech**
Hola 👋 

I would like to request a feature for this kind of code

```ruby
router = Hanami::Router.new do
  namespace 'api' do
    mount Api::Users, at: '/users'
    mount Api::Posts, at: '/posts'
  end
end

puts router.inspector.to_s

##
# Before
# Name Method     Path                 Action                        
#
#                 /users               #<Proc@.../endpoint_resolver.rb:189 (lambda)>
#                 /posts               #<Proc@.../endpoint_resolver.rb:189 (lambda)>

##
# After
# Name Method     Path                 Action                        
#
#                 /api/users           #<Proc@.../endpoint_resolver.rb:189 (lambda)>
#                 /api/posts           #<Proc@.../endpoint_resolver.rb:189 (lambda)>
```

It was a bit confusing, that namespace doesn't impact mounts.

### CODE REVIEW

**Comment by Strech** [comment]
@jodosha copy, will adjust pull request

File: lib/hanami/routing/namespace.rb [hunk 1]
```diff
@@ -81,6 +81,12 @@ def redirect(path, options = {}, &endpoint)
         super(@name.join(path), options.merge(to: @name.join(options[:to])), &endpoint)
       end
 
+      # @api private
+      # @since 1.0.1
```
Comment (Strech) [inline lib/hanami/routing/namespace.rb]: @jodosha I'm not sure about this number

**Comment by Strech** [comment]
@jodosha Tests are ported to RSpec

File: lib/hanami/routing/namespace.rb (hunk 1 above)
Comment (cllns) [reply lib/hanami/routing/namespace.rb]: Use `x.x.x` for now 👍

**Comment by AlfonsoUceda** [comment]
@Strech thank you so much for this!! can you amend this https://github.com/hanami/router/pull/150#discussion_r143799238? To me it's ok 👍

File: lib/hanami/routing/namespace.rb (hunk 1 above)
Comment (Strech) [reply lib/hanami/routing/namespace.rb]: Done

**Comment by Strech** [comment]
Last changes: Update private since to `x.x.x`

File: spec/integration/hanami/router/namespace_spec.rb [hunk 2]
```diff
@@ -411,5 +411,19 @@
...
     end
+
+    describe 'mount' do
+      before do
+        @router.namespace 'api' do
+          mount Backend::App, at: '/backend'
+        end
+      end
+
+      [ 'get', 'post', 'delete', 'put', 'patch', 'trace', 'options', 'link', 'unklink' ].each do |verb|
```
Comment (davydovanton) [inline spec/integration/hanami/router/namespace_spec.rb]: just style question, maybe `%w[]`?

Comment (jodosha) [reply spec/integration/hanami/router/namespace_spec.rb]: We don't enforce style yet on this code base.

**Comment by jodosha** [comment]
@Strech This looks good now. Merging it. It will be part of the next beta. Thanks!

**Pull request merged**